confmap/provider.go

```diff
@@ -165,9 +165,6 @@ func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (*Retrieved
 //   - []any;
 //   - map[string]any;
 func NewRetrieved(rawConf any, opts ...RetrievedOption) (*Retrieved, error) {
-	if err := checkRawConfType(rawConf); err != nil {
```

Comment:
- The env var is `2023-03-20T03:17:55.432328Z` without quotes
- This `"${env:TIMESTAMP}"` is exactly what we have now and it's failing with the error
- The value is `time.Time`. I was basing the assumption we allow any types following this PR https://github.com/open-telemetry/opentelemetry-collector/pull/10800. Happy to add it as a type if that's the agreed fix.

Reply:
@TylerHelmuth I've changed it to add `time.Time` as a type accepted in the check vs removing the check completely

Reply:
As a sanity check, what happens today if the env var includes the `""`, such as `"2023-03-20T03:17:55.432328Z"`?

Reply:
Yes that one results in `"2023-03-20T03:17:55.432328Z"`. This fix is primarily handling the case where a time string is provided without quotes and unexpectedly encounters an error (which is what's happening in our case since this can be a user-provided environment variable)

Reply:
Cool